File: android/src/main/kotlin/com/emarsys/emarsys_sdk/mapper/ProductsMapper.kt
package com.emarsys.emarsys_sdk.mapper

import com.emarsys.predict.api.model.Product

class ProductsMapper {
    fun map(products: List<Product>): List<Map<String, Any>> {
        return products.map { element ->
            val resultMap = mutableMapOf<String, Any>(
                "productId" to element.productId,
                "linkUrl" to element.linkUrl
            )
            resultMap["available"] =  element.available ?: false
            resultMap.toMap()
        }
    }
}

File: ios/Classes/Wrapper/Mappers/ProductsMapper.swift
//
//  ProductsMapper.swift
//  emarsys_sdk
//
//  Created by Doris on 2021/9/1.
//

import Foundation
import EmarsysSDK

class ProductsMapper: Mappable {
    
    typealias Input = [EMSProduct]
    typealias Output = [[String: Any]]?
    
    func map(_ input: [EMSProduct]) -> [[String : Any]]? {
        let productDicts: [[String: Any]]? = input.map { element in
            var productDict = [String: Any]()
            productDict["productId"] = element.productId
            productDict["linkUrl"] = element.linkUrl.absoluteString
            productDict["available"] = element.available
            return productDict
        }
        return productDicts
    }
    
}
